insights/client/core_collector.py

```diff
@@ -51,7 +52,8 @@ def run_collection(self, conf, rm_conf, branch_info, blacklist_report):
 
         manifest = collect.default_manifest
         if hasattr(self.config, 'manifest') and self.config.manifest:
-            manifest = self.config.manifest
+            with open(self.config.manifest, 'r') as f:
```

Comment:
When `--collector` is used it sets the variable to the content of a content manifest. In this case, this line will fail as config.manifest is not a file.
I'd suggest to check if config.app is not None before open the manifest as a file. 

Also to check if config.manifest is a file, we would want to handle the exception if the manifest is wrong specified.

Reply:
Does it make sense to catch the Exception if we don't have any action to take? Shouldn't we let the Exception go unhandled, so that callers of `run_collection` can catch it and handle it accordingly?